synapse/util/logcontext.py

```diff
@@ -225,7 +234,7 @@ def __init__(self, name=None, parent_context=None, request=None):
         # became active.
         self.usage_start = None
 
-        self.main_thread = threading.current_thread()
```

Comment:
for clarity: this isn't a necessary part of the fix, but something I came across while I was trying to figure out what was going on here. It seems more sensible to compare ints than objects.